### PR TITLE
Quality of life fixes and new autoscaling modes

### PR DESCRIPTION
**Pull request by jasongaunt**
Fixes:
* Timestamps on graph missing leading zero
* Graph lines clipping off extreme ends of graph
* Initial graph values being 0 causing a line to be drawn to 0

New features:
* New autoscaling modes allowing fine-grain control over min and max behaviour
* Units of measurement now drawn on graph axis markers with HTML entity parsing

### CODE REVIEW

File: picograph.js [hunk 1]
```diff
@@ -153,11 +175,12 @@ class Graph {
 
 
         /* Draw horizontal scale */
+        var entityDecode = document.createElement('textarea'); entityDecode.innerHTML = this.unit;
```
Comment (RainingComputers) [inline picograph.js]: Maybe `let` instead of `var`? Or is there a good reason for this being a `var`?

File: README.md [hunk 2]
```diff
@@ -63,14 +63,16 @@ function createGraph(canvasID, labels, unit, labelDivID, intervalSize,
...
+argument is only for initial value. Set to `null` to automatically determine.
 + **minVal**: Approximate minimum value. Picograph can autoscale, this 
 argument is only for initial value.
 + **vlines**: Show vertical lines.
 + **timestamps**: Show timestamps.
 + **scalesteps**: Number of scale lines to draw on the graph.
 + **vlinesFreq**: Vertical lines and timestamps will be drawn every `vlinesFreq*intervalSize`. Increase this
 if the vertical lines are too crowded.
++ **autoScaleMode**: When 0, minVal and maxVal become absolutes. When 1, autoscale to fit peaks only.
+When 2, graph scaling constanly adjusts to fit visible data.
```
Comment (RainingComputers) [inline README.md]: **constantly** instead of **constanly** typo?

**Comment by RainingComputers** [comment]
Thanks for contributing! 😄

**Comment by jasongaunt** [comment]
Thanks @RainingComputers - Typo fixed and entityDecode variable swapped to local 👍

**Pull request merged**